.github/workflows/cloud_build_failure_reporter.yml

```diff
@@ -0,0 +1,162 @@
+                      await github.rest.issues.createComment({
+                        ...context.repo,
+                        issue_number: prevIssue.number,
+                        body: txt
+                      });
+                    }
+                  };
+
+                  // Find status of check runs.
+                  // We will find check runs for each commit and then filter for the periodic.
+                  // Checks API only allows for ref and if we use main there could be edge cases where
+                  // the check run happened on a SHA that is different from head.
+                  const commits = await github.paginate(github.rest.repos.listCommits, {
+                    ...context.repo
+                  });
+
+                  const relevantChecks = [];
+                  for (const commit of commits) {
+                    console.log(
+                      `checking runs at ${commit.html_url}: ${commit.commit.message}`
+                    );
+                    const checks = await github.rest.checks.listForRef({
+                      ...context.repo,
+                      ref: commit.sha
+                    });
+
+                    // Iterate through each check and find matching names
+                    for (const check of checks.data.check_runs) {
+                      for (const testName of testNameFound.keys()) {
+                        if (testNameFound.get(testName) === true){
+                          //skip if a check is already found for this name
+                          continue; 
+                        }
+                        if (check.name == testName) {
+                          relevantChecks.push(check);
+                          testNameFound.set(testName, true);
+                        }
+                      }
+                    }
+
+                  // Handle each relevant check
+                  for (const check of relevantChecks) {
+                      if (
+                          check.status === 'completed' &&
+                          check.conclusion === 'success'
+                      ) {
+                          updateIssues(
+                              `[Passing periodic](${check.html_url}) at ${commit.html_url}.`
+                          );
+                      } else if (check.status === 'in_progress') {
+                          console.log(
+                              `Check is pending ${check.html_url} for ${commit.html_url}. Retry again later.`
+                          );
+                      } else {
+                          createOrCommentIssue(
+                              `[Cloud Build Failure Reporter] test failing for **${check.name}**`,
+                              `Cloud Build Failure Reporter found test failure for **${check.name}** ([link to check](${check.html_url})) at [link to commit](${commit.html_url}). Please fix the error and then close the issue after the **${check.name}** test passes.`
```

Comment:
Instead of using "link to" just add wrap the check name and commit sha, respectively.

Reply:
updated